File: 0148-sort-list/0148-sort-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) 
    {
        vector<int> v;
        ListNode *curr = head;
        while(curr!=NULL)
        {
            v.push_back(curr->val);
            curr=curr->next;
        }
        sort(v.begin(),v.end());
        ListNode *curr2 = head;
        int k=0;
        while(curr2!=NULL && k<v.size())
        {
            curr2->val = v[k];
            curr2 = curr2->next;
            k++;
        }
        return head;
    }
};

File: 0148-sort-list/0148-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        if(head == null){
            return null;
        }
        ArrayList<Integer> ar = new ArrayList<Integer>();
        ListNode curr = head;

        while(curr!=null){
            ar.add(curr.val);
            curr = curr.next;
        }
        Collections.sort(ar);
        int p=0;
        ListNode temp = head;
        while(temp!=null && p<ar.size()){
            temp.val = ar.get(p);
            temp = temp.next;
            p++;
        }
        return head;

    }
}